

## Overview-ScatterPlot


| [ScatterPlot-sp1](#sp1)                                      | [ScatterPlot-sp2](#sp2)                                      | [ScatterPlot-sp3](#sp3)                                      |
| ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
| <img src="..\..\images_for_markdown\Common_plots\scatterPlot\scatterPlot_style1.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\scatterPlot\scatterPlot_styple2.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\scatterPlot\scatterPlot_fit.png" style="zoom: 25%;" /> |
| **[ScatterPlot-sp4](#sp4)**                                  | **[ScatterPlot-sp5](#sp5)**                                  | **[ScatterPlot-sp6](#sp6)**                                          |
| <img src="..\..\images_for_markdown\Common_plots\scatterPlot\ordered_scatterplot_baseR.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\scatterPlot\ordered_scatterplot_ggplot2.png" style="zoom:25%;" /> |                                                              |



## Overview-BarPlot

| [BarPlot-bp1](#bp1)                                                  | [BarPlot-bp2](#bp2)                                                  | [BarPlot-bp3](#bp3)                                                  |
| ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
| <img src="..\..\images_for_markdown\Common_plots\barplot\barplot_sided.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\barplot\barplot_style2.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\barplot\barplot_style3.png" style="zoom: 25%;" /> |
| **[BarPlot-bp4](#bp4)**                                              | **[BarPlot-bp5](#bp5)**                                              | **[BarPlot-bp6](#bp6)**                                              |
| <img src="..\..\images_for_markdown\Common_plots\barplot\barplot_symmetry.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\barplot\ordered_barplot.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\barplot\ordered_barplot_stack.png" style="zoom: 25%;" /> |



## Overview-BoxPlot

| [BoxPlot-bxp1](#bxp1)                                                 | [BoxPlot-bxp2](#bxp2)                                                 | [BoxPlot-bxp3](#bxp3)                                                 |
| ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
| <img src="..\..\images_for_markdown\Common_plots\boxplot\boxplot_multi.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\boxplot\ordered_boxplot.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\boxplot\ordered_boxplot_grid.png" style="zoom: 25%;" /> |
| **[BoxPlot-bxp4](#bxp4)**                                             | **[BoxPlot-bxp5](#bxp5)**                                             | **[BoxPlot-bxp6](#bxp6)**                                             |
| <img src="..\..\images_for_markdown\Common_plots\boxplot\ordered_boxplot_style2.png" style="zoom: 25%;" /> |                                                              |                                                              |



## Overview-HeatMap

| [HeatMap-hm1](#hm1)                                                  | [HeatMap-hm2](#hm2)                                                  | [HeatMap-hm3](#hm3)                                                  |
| ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
| <img src="..\..\images_for_markdown\Common_plots\heatmap\heatmap_style1.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\heatmap\heatmap_style2.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\heatmap\heatmap_style3.png" style="zoom: 25%;" /> |
| **[HeatMap-hm4](#hm4)**                                              | **[HeatMap-hm5](#hm5)**                                              | **[HeatMap-hm6](#hm6)**                                              |
| <img src="..\..\images_for_markdown\Common_plots\heatmap\heatmap_style4.png" style="zoom: 25%;" /> |                                                              |                                                              |



## Overview-LinePlot

| [LinePlot-lp1](#lp1)                                                 | [LinePlot-lp2](#lp2)                                                 | [LinePlot-lp3](#lp3) |
| ------------------------------------------------------------ | ------------------------------------------------------------ | ------------ |
| <img src="..\..\images_for_markdown\Common_plots\linePlot\dot_line_plot.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\linePlot\forest_line_plot.png" style="zoom: 25%;" /> |              |



## Overview-ForestPlot

| [ForestPlot-fp1](#fp1)                                               | [ForestPlot-fp2](#fp2)                                               | [ForestPlot-fp3](#fp3) |
| ------------------------------------------------------------ | ------------------------------------------------------------ | -------------- |
| <img src="..\..\images_for_markdown\Common_plots\forestPlot\forestplot_baseR.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\forestPlot\forestplot_ggplot2.png" style="zoom: 25%;" /> |                |



## Overview-PiePlot

| [PiePlot-pp1](#pp1)                                                  | [PiePlot-pp2](#pp2) | [PiePlot-pp3](#pp3) |
| ------------------------------------------------------------ | ----------- | ----------- |
| <img src="..\..\images_for_markdown\Common_plots\PiePlot\scatterPie_tree_plot.png" style="zoom: 25%;" /> |             |             |



## Overview-SurvivalPlot

| [SurvivalPlot-sup1](#sup1)                                            | [SurvivalPlot-sup2](#sup2) | [SurvivalPlot-sup3](#sup3) |
| ------------------------------------------------------------ | ----------------- | ----------------- |
| <img src="..\..\images_for_markdown\Common_plots\survivalPlot\survival_curve.png" style="zoom: 25%;" /> |                   |                   |



## Overview-CombPlot

| [CombPlot-cp1](#cp1)                                                 | [CombPlot-cp2](#cp2)                                                 | [CombPlot-cp3](#cp3) |
| ------------------------------------------------------------ | ------------------------------------------------------------ | ------------ |
| <img src="..\..\images_for_markdown\Common_plots\combPlot\ordered_barplot_comb.png" style="zoom: 25%;" /> | <img src="..\..\images_for_markdown\Common_plots\combPlot\scatterPlot_barplot_comb.png" style="zoom: 25%;" /> |              |



## ScatterPlot



### sp1

<img src="..\..\images_for_markdown\Common_plots\scatterPlot\scatterPlot_style1.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. 阿里云盘，[mutsig_fit.R](https://www.aliyundrive.com/drive/folder/5fb620ec327d6290953342bbb1cf37392d746f2f)
2. PC端，[mutsig_fit.R](F:\common_R_functions\mutsig_fit.R)

```R
sam_sse_plot <- function(sam_burden_sse, sse_cut_off = 0.2, mut_cut_off = NA, mut_logscale = TRUE, sam_col = "Sample", sse_col = "SSE", mut_count_col = "Snv_burden", plot_path){
  # sam_sse_data：样本的突变数目及SSE数据
  # sse_cut_off：SSE的阈值特殊标记
  # mut_cut_off：突变数目阈值特殊标记
  # mut_logscale：是否对突变数目进行log10转化
  # sam_col：样本列名
  # sse_col：SSE列名
  # mut_count_col：突变数目列名
  # plot_path：绘图存放路径
  
  # sam_burden_sse <- skcm_mutSig_mer_list$sam_burden_sse
  # sse_cut_off <- 0.2
  # mut_cut_off <- 2153
  # mut_logscale <- TRUE
  # sam_col <- "Sample"
  # sse_col <- "SSE"
  # mut_count_col <- "Snv_burden"
  
  sam_burden_sse <- select(sam_burden_sse, Sample = sam_col, SSE = sse_col, Snv_burden = mut_count_col)
  
  plot <- ggplot(sam_burden_sse, aes(x = Snv_burden, y = SSE)) + 
    geom_point(data = filter(sam_burden_sse, SSE > sse_cut_off), stat = "identity", col = "red", pch = 8) + 
    geom_point(data = filter(sam_burden_sse, SSE <= sse_cut_off), stat = "identity", col = "gray70") +
    labs(x = "SNV Count", y = "deconstructSigs SSE")
  
  if(!is.na(mut_cut_off)){
    plot <- plot +
      geom_point(data = filter(sam_burden_sse, Snv_burden >= mut_cut_off), stat = "identity", col = "black", pch = 17)
  }
  
  if(mut_logscale){
    plot <- plot + scale_x_log10()
  }
  
  pdf(str_c(plot_path, "/sam_sse_scatterplot.pdf"))
  print(plot)
  dev.off()
}
```



### sp2

<img src="..\..\images_for_markdown\Common_plots\scatterPlot\scatterPlot_styple2.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. 阿里云盘，[mutSig_related_gene.R](https://www.aliyundrive.com/drive/folder/5fb620ec327d6290953342bbb1cf37392d746f2f)
2. PC端，[mutSig_related_gene.R](F:\common_R_functions\mutSig_related_gene.R)

```R
gs_p_scatter_plot <- function(medep_p_data, which_signature, min_label_value){
  # medep_p_data：函数MergePmediff所得融合数据
  # which_signature：需绘制的signature，为单数
  # min_label_value：为大于等于该值的点进行基因标记
  
  # medep_p_data <- skcm_medep_p_data
  # which_signature <- "Signature.7"
  # min_label_value <- 2.55
  
  plot_data <- medep_p_data %>% filter(Signature %in% which_signature)
  plot <- plot_data %>% 
    ggplot(aes(x = median_ep_diff, y = log_permu_p)) +
    geom_point(data = filter(plot_data, log_permu_p < min_label_value), aes(size = log_permu_p), alpha = 0.8) + 
    geom_point(data = filter(plot_data, log_permu_p >= min_label_value), aes(size = log_permu_p), alpha = 0.8, col = "red") + 
    geom_text(aes(median_ep_diff, log_permu_p, label = Gene), data = filter(plot_data, log_permu_p >= min_label_value), size = 2.5, fontface = "italic", col = "blue", nudge_y = -0.25) + 
    labs(x = str_c("Median difference of signature", str_extract(which_signature, "[\\d]"), "mutations", sep = ' '), y = "-log10(permutation P value)") + 
    theme(legend.position = "none", axis.title = element_text(size = 10))
}
```



### sp3

<img src="..\..\images_for_markdown\Common_plots\scatterPlot\scatterPlot_fit.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[SKCM_mutSig_cli_association.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section1_mutSig_fit_characterization\code\SKCM_mutSig_cli_association.R)

```R
skcm_sig5_age_plot <- ggplot(filter(skcm_ms_plot_data, Signature %in% "Signature.5"), aes(age_at_initial_pathologic_diagnosis, Exposure)) +
  geom_point(col = "darkgreen") + 
  geom_smooth(method = "lm", col = "darkred") + 
  labs(x = "Age of cancer diagnosis (years)", y = "Number of mutations", title = "Signature 5") + 
  annotate("text", x = 20, y = 600, label = "P value = 9.43E-05", vjust = "inward", hjust = "inward") +
  coord_cartesian(ylim = c(0, 600)) +
  theme(plot.title = element_text(hjust = 0.5), axis.title.x = element_text(size = 10), axis.title.y = element_text(size = 10))
```




### sp4

![](..\..\images_for_markdown\Common_plots\scatterPlot\ordered_scatterplot_baseR.png)


#### 原始数据

| cohort | Tumor_Sample_Barcode | total | site                    |
| ------ | -------------------- | ----- | ----------------------- |
| ACC    | TCGA-OR-A5KB         | 1841  | Adrenal Gland Carcinoma |
| ACC | TCGA-PK-A5HB | 975 | Adrenal Gland Carcinoma |

#### 语言

R

#### 绘图模块或包

base

#### 代码

1. 阿里云盘，[mutSig_related_gene.R](https://www.aliyundrive.com/drive/folder/5fb620ec327d6290953342bbb1cf37392d746f2f)
2. PC端，[mutSig_related_gene.R](F:\common_R_functions\mutSig_related_gene.R)

```R

# 函数：绘制癌症的突变数目散点图 ---------------------------------------------------------

cohort_mut_plot <- function(cohort_mut_data, cancer_type_column, sample_column, mut_count_column, col = NULL, bg_col = c("#EDF8B1", "#2C7FB8"), median_col = "red", logscale = TRUE, plot_path){
  # cohort_mut_data：癌症中的样本突变数目数据
  # cancer_type_column：癌症类型列名
  # sample_column：样本列名
  # mut_count_column：样本突变数目列名
  # col：为每个癌症类型指定颜色，若NULL则为灰色
  # bg_col：散点图的背景颜色
  # median_col：散点图中值位点颜色
  # logscale：是否对突变数目进行log10转换
  # plot_path：绘图存放路径
  
  # cohort_mut_data <- tcga_cohort_mut
  # cancer_type_column <- "cohort"
  # sample_column <- "Tumor_Sample_Barcode"
  # mut_count_column <- "total"
  # col <- NULL
  # bg_col <- c("#EDF8B1", "#2C7FB8")
  # median_col <- "red"
  # logscale <- TRUE
  
  cohort_mut_data <- select(cohort_mut_data, cancer = cancer_type_column, sample = sample_column, mut_count = mut_count_column)
  
  cohort_mut_med <- cohort_mut_data %>%
    group_by(cancer) %>%
    summarise(case = n(), mut_med = median(mut_count), mut_med_log10 = log10(mut_med)) %>%
    arrange(mut_med)
  
  cohort_mut_data$cancer <- factor(x = cohort_mut_data$cancer, levels = cohort_mut_med$cancer) 
  cohort_mut_lt <- split(cohort_mut_data, cohort_mut_data$cancer)
  
  # list中存放每个癌症的排序好的突变数据，设定递增的x轴坐标(0-1之间，1-2之间...)
  
  plot_dat <- lapply(1:length(cohort_mut_lt), function(i) {
    x <- cohort_mut_lt[[i]] %>%
      arrange(mut_count) %>%
      mutate(scale_x_value = seq(i - 1, i, length.out = n()))
  })
  names(plot_dat) = names(cohort_mut_lt)
  
  # 限制y轴长度 ------------------------------------------------------------------
  
  if (logscale) {
    y_lims = range(log10(bind_rows(plot_dat)$mut_count))
  }else {
    y_lims = range(bind_rows(plot_dat)$mut_count)
  }
  y_max = ceiling(max(y_lims))
  y_min = floor(min(y_lims))
  y_lims = c(y_min, y_max)
  y_at = pretty(y_lims) # y轴虚线分割线
  
  pdf(str_c(plot_path, "/mut_scatter_plot.pdf"))
  par(mar = c(4, 4, 2, 1))
  
  # 散点图的框架 ------------------------------------------------------------------
  
  plot(NA, NA, xlim = c(0, length(plot_dat)), ylim = y_lims, xlab = NA, ylab = NA, bty = "n", xaxt = "n", las = 2)
  rect(xleft = seq(0, length(plot_dat) - 1, 1), ybottom = y_min, xright = seq(1, length(plot_dat), 1), 
       ytop = y_max, col = grDevices::adjustcolor(col = bg_col, alpha.f = 0.2), border = NA)
  abline(h = pretty(y_lims), lty = 2, col = "gray70")
  
  # 散点 ----------------------------------------------------------------------
  
  if (is.null(col)) {
    col <- rep("gray70", length(plot_dat))
    names(col) <- names(plot_dat)
  }
  
  lapply(1:length(plot_dat), function(i) {
    x = plot_dat[[i]]
    if (logscale) {
      points(x$scale_x_value, log10(x$mut_count), pch = 16, cex = 0.4, col = col[names(plot_dat)[i]])
    }else {
      points(x$scale_x_value, x$mut_count, pch = 16, cex = 0.4, col = col[names(plot_dat)[i]])
    }
  })
  
  # x轴和y轴标签 -----------------------------------------------------------------
  
  axis(side = 1, at = seq(0.5, length(plot_dat) - 0.5, 1), labels = names(plot_dat), las = 2, tick = FALSE, line = -1)
  axis(side = 3, at = seq(0.5, length(plot.dat) - 0.5, 1), labels = cohort_mut_med$case, las = 2, tick = FALSE, line = -1.2, font = 3)
  
  if (logscale) {
    mtext(text = "TMB (log10)", side = 2, line = 2)
  }else {
    mtext(text = "TMB", side = 2, line = 2)
  }
  
  # 中值线 ---------------------------------------------------------------------
  
  if (logscale) {
    lapply(1:nrow(cohort_mut_med), function(i) {
      x_seq <- quantile(c(i -1, i), seq(0.1, 1, 0.1))
      segments(x0 = x_seq["30%"], x1 = x_seq["70%"], y0 = cohort_mut_med$mut_med_log10[i], y1 = cohort_mut_med$mut_med_log10[i], col = median_col)
    })
  }else {
    lapply(1:nrow(cohort_mut_med), function(i) {
      x_seq <- quantile(c(i -1, i), seq(0.1, 1, 0.1))
      segments(x0 = x_seq["30%"], x1 = x_seq["70%"], y0 = cohort_mut_med$mut_med_log10[i], y1 = cohort_mut_med$mut_med_log10[i], col = median_col)
    })
  }
  
  dev.off()
  
}

# 测试 ----------------------------------------------------------------------

# tcga_cohort_mut <- read_tsv("/Users/jianlongliao/tcga_cohort_plot/tcga_cohort.txt")
# tcga_cohort_mut_plot <- cohort_mut_plot(tcga_cohort_mut, "cohort", "Tumor_Sample_Barcode", "total", plot_path = "/Users/jianlongliao/tcga_cohort_plot")

```



### sp5

![](..\..\images_for_markdown\Common_plots\scatterPlot\ordered_scatterplot_ggplot2.png)

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码


```R
library(tidyverse)

test_data <- read.table("/Users/jianlongliao/tcga_cohort_plot/tcga_cohort.txt", sep = "\t", header = TRUE, stringsAsFactors = FALSE)

plot_data <- test_data %>%
  group_by(cohort) %>%
  arrange(total, .by_group = TRUE) %>%
  mutate(x_order = 1:length(total), med_count = median(total), med_order = median(x_order)) %>%
  ungroup() %>%
  mutate(cohort = fct_reorder(cohort, total), med_base = max(med_order), adjust_x_order = x_order + med_base - med_order)

med_count_data <- plot_data %>% 
  distinct(cohort, med_count, med_base) %>%
  mutate(med_base_minus = med_base - 300, med_base_plus = med_base + 300)

ggplot(plot_data, aes(adjust_x_order, total)) +
  geom_point(aes(col = cohort), size = 0.1) +
  facet_grid(.~cohort) +
  scale_y_log10() +
  labs(x = NULL, y = "Number of mutations") +
  geom_segment(aes(x = med_base_minus, y = med_count, xend = med_base_plus, yend = med_count), col = "red", data = med_count_data) +
  theme_classic() +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), 
        legend.position = "none", strip.background = element_blank(), strip.text = element_text(angle = 90))
ggsave("/Users/jianlongliao/tcga_cohort_plot/scatter_order_ggplot.pdf", width = 10, height = 6)

```



## Barplot

### bp1

<img src="..\..\images_for_markdown\Common_plots\barplot\barplot_sided.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2, ggsci

#### 代码

1. PC端，[mutSig_subtype_multi_omics_analysis.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section3_mutSig_dynamics_subtype\code\mutSig_subtype_multi_omics_analysis.R)

```R
plot_data <- mut_freq_test %>%
  pivot_longer(NRAS:CDKN2A, names_to = "Hugo_Symbol", values_to = "status") %>%
  group_by(Osc_cluster, Hugo_Symbol) %>%
  summarise(mut_freq = mean(status)) %>%
  filter(Hugo_Symbol %in% c("COL5A1", "BRAF"))

ggplot(plot_data, aes(Hugo_Symbol, mut_freq, fill = Osc_cluster)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_fill_nejm() +
  labs(x = NULL, y = "Proportion of samples", fill = NULL) +
  scale_y_continuous(expand = c(0, 0), limits = c(NA, 0.8)) +
  scale_x_discrete(labels = c(BRAF = "BRAF mutant", COL5A1 = "COL5A1 mutant")) +
  theme_bw() +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_text(face = "bold", size = 10), axis.title.y = element_text(face = "bold", size = 10))
ggsave("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section3_mutSig_dynamics_subtype/subtype_multi_omics/driver_freq_subtype_barplot.pdf")
```



### bp2

<img src="..\..\images_for_markdown\Common_plots\barplot\barplot_style2.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[clonality_method_determination.R](G:\xiaolonggeのhome\已完成课题\克隆性分型课题\改稿\投Neuropathology and Applied Neurobiology\负责问题\code\submit\clonality_method_determination.R)

```R
# driver clonal fraction barplot

ggplot(filter(test_data, !clonality_method %in% "ccf.0.8", Hugo_Symbol %in% c("Drivers", "IDH1", "PIK3CA", "TP53")), aes(clonality_method, fill = clonality)) +
  geom_bar(position = "fill") +
  labs(x = NULL, y = "Proportion of mutations", fill = NULL) +
  scale_fill_manual(values = c("#D45658", "#3A84BB")) +
  scale_x_discrete(breaks = c("CI95.timing", "prob.clonal.timing"), labels = c("CI95 based", "Prob based")) +
  scale_y_continuous(expand = c(0, 0)) +
  theme_classic() +
  theme(axis.line.x = element_blank(), axis.ticks.x = element_blank(), strip.background = element_blank(), axis.text.x = element_text(angle = 45, vjust = 0.5, hjust = 0.5)) +
  facet_grid(. ~ Hugo_Symbol)
ggsave("/Users/jianlongliao/xiaolonggeのhome/已完成课题/克隆性分型课题/改稿/投Neuropathology and Applied Neurobiology/负责问题/section_1/clonality_method_frac_barplot.pdf", width = 6, height = 5)
```



### bp3

<img src="..\..\images_for_markdown\Common_plots\barplot\barplot_style3.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[total_mutsig_fit.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section1_mutSig_fit_characterization\code\total_mutsig_fit.R)

```R
# SKCM突变signature的碱基改变模式条形图 -----------------------------------------------

tri_context_level <- colnames(skcm_cosmic_sig_profile)
sig_level <- rownames(skcm_cosmic_sig_profile)

skcm_cosmic_sig_tib <- as_tibble(skcm_cosmic_sig_profile) %>%
  mutate(Signature = rownames(skcm_cosmic_sig_profile)) %>%
  pivot_longer(-Signature, names_to = "tri_context", values_to = "fraction") %>%
  mutate(base_change = str_sub(tri_context, 3, 5))
skcm_cosmic_sig_tib$tri_context <- factor(skcm_cosmic_sig_tib$tri_context, levels = tri_context_level)
skcm_cosmic_sig_tib$Signature <- factor(skcm_cosmic_sig_tib$Signature, levels = sig_level)

pdf("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section1_mutSig_fit_characterization/mutSig_fit_result/skcm_mutSig_profile_plot.pdf")
ggplot(skcm_cosmic_sig_tib, aes(x = tri_context, y = fraction, fill = base_change)) + 
  geom_bar(stat = "identity") + 
  coord_cartesian(ylim = c(0, 0.2)) + 
  facet_grid(Signature ~ .) + 
  labs(x = "", y = "Fraction of mutations") + 
  guides(fill = guide_legend(nrow = 1)) + 
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5, size = 4), axis.ticks.x = element_blank(), legend.position = 'top', legend.title = element_blank()) + 
  scale_fill_nejm()
dev.off()
```



### bp4

<img src="..\..\images_for_markdown\Common_plots\barplot\barplot_symmetry.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. 阿里云盘，[mutSig_related_gene.R](https://www.aliyundrive.com/drive/folder/5fb620ec327d6290953342bbb1cf37392d746f2f)
2. PC端，[mutSig_related_gene.R](F:\common_R_functions\mutSig_related_gene.R)

```R
permu_p_compare_barplot <- function(medep_p_data1, medep_p_data2, which_signature, line_value){
  
  # line_value：辅助线阈值
  
  # medep_p_data1 <- early_medep_p_data$ep_mediff_p_stat
  # medep_p_data2 <- late_medep_p_data$ep_mediff_p_stat
  # which_signature <- "Signature.5"
  # line_value <- 2.5
  
  medep_p_data <- medep_p_data1 %>%
    full_join(medep_p_data2, by = c("Signature", "Gene")) %>%
    filter(Signature %in% which_signature) %>%
    mutate(Early = log_permu_p.x, Late = -log_permu_p.y) %>%
    select(Signature, Gene, Early, Late) %>%
    pivot_longer(Early:Late, names_to = "stage", values_to = "log_permu_p")
  
  p_plot <- ggplot(medep_p_data, aes(x = Gene, y = log_permu_p, fill = stage)) +
    geom_bar(stat = "identity") +
    labs(x = NULL, y = str_c("-log10(permutation P value) of signature ", str_extract(which_signature, "[\\d]")), fill = NULL) +
    coord_flip() +
    geom_hline(yintercept = min_label_value, col = "red", linetype = "dashed") +
    geom_hline(yintercept = -min_label_value, col = "red", linetype = "dashed") +
    scale_fill_nejm() +
    theme(legend.position = "top", axis.text.y = element_text(size = 8, face = "italic"), axis.title.x = element_text(size = 8))
}
```



### bp5

<img src="..\..\images_for_markdown\Common_plots\barplot\ordered_barplot.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[mut_clonality_stat_analysis.R](G:\xiaolonggeのhome\已完成课题\克隆性分型课题\改稿\投Neuropathology and Applied Neurobiology\负责问题\code\submit\mut_clonality_stat_analysis.R)

```R
# sam_mut_num_barplot

sam_mut_order <- glioma_mut_clonality_data %>%
  count(patient, sort = TRUE)

sam_mut_num_barplot_data <- glioma_mut_clonality_data %>%
  mutate(patient = factor(patient, levels = sam_mut_order$patient), 
         IDH.codel.subtype = factor(IDH.codel.subtype, levels = c("IDHwt", "IDHmut-non-codel", "IDHmut-codel")))

barplot_basic <- ggplot(sam_mut_num_barplot_data, aes(patient, fill = CI95.timing)) +
  geom_bar() +
  labs(x = NULL, y = "Number of mutations", fill = NULL) +
  scale_fill_nejm() +
  theme_classic() +
  scale_x_discrete(expand = c(0.015, 0)) +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), strip.background = element_blank()) +
  facet_grid(. ~ IDH.codel.subtype, space = "free_x", scales = "free_x")

barplot_part1 <- barplot_basic +
  scale_y_continuous(expand = c(0.015, 0)) +
  coord_cartesian(ylim = c(0, 540)) +
  theme(strip.text = element_blank(), legend.position = "bottom")

barplot_part2 <- barplot_basic +
  coord_cartesian(ylim = c(680, 850)) +
  labs(y = "") +
  scale_y_continuous(breaks = c(600, 800, 100)) +
  theme(axis.line.x = element_blank(), legend.position = "none")

comb_plot <- gridExtra::grid.arrange(barplot_part2, barplot_part1, heights = c(1/5, 4/5), ncol = 1, nrow = 2)

ggsave("/Users/jianlongliao/xiaolonggeのhome/已完成课题/克隆性分型课题/改稿/投Neuropathology and Applied Neurobiology/负责问题/section_1/sam_mut_num_barplot.pdf", comb_plot, width = 10, height = 4)
```



### bp6

<img src="..\..\images_for_markdown\Common_plots\barplot\ordered_barplot_stack.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[mutSig_subtype_feature_association_analysis.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section3_mutSig_dynamics_subtype\code\mutSig_subtype_feature_association_analysis.R)

```R
ggplot(cluster_comb_data, aes(Osc_cluster, fill = genome_doubling)) +
  geom_bar(position = position_fill(reverse = TRUE)) +
  labs(x = NULL, y = "Proportion of samples", fill = "Genome doubling") +
  theme(axis.text.x = element_text(face = "bold", size = 10), axis.ticks.x = element_blank(), axis.title.y = element_text(face = "bold", size = 10)) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values = c("#80796B", "#0072B5", "#BC3C29")) +
  guides(fill = guide_legend(reverse = TRUE))
ggsave("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section3_mutSig_dynamics_subtype/subtype_feature_association_analysis/subtype_gd_boxplot.pdf")
```



## BoxPlot

### bxp1

<img src="..\..\images_for_markdown\Common_plots\boxplot\boxplot_multi.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2, ggsci

#### 代码

1. PC端，[mutSig_subtype_feature_association_analysis.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section3_mutSig_dynamics_subtype\code\mutSig_subtype_feature_association_analysis.R)

```R
cluster_comb_data %>% select(Sample:Osc_cluster, Clonal:Subclonal, `Total SNV`  = snv_count) %>%
  pivot_longer(Clonal:`Total SNV`, names_to = "Type", values_to = "Count") %>%
  ggplot(aes(Type, Count)) +
  geom_boxplot(aes(fill = Osc_cluster), position = position_dodge(width = 0.9), outlier.color = NA, alpha = 0.8, show.legend = FALSE) +
  geom_point(aes(color = Osc_cluster), position = position_jitterdodge(jitter.width = 0.1, dodge.width = 0.9)) +
  labs(x = NULL, y = "Number of mutations", color = NULL, fill = NULL) +
  theme(axis.text.x = element_text(face = "bold", size = 10), axis.ticks.x = element_blank(), axis.title.y = element_text(face = "bold", size = 10)) +
  scale_color_nejm() +
  scale_fill_manual(values = rep("#80796B", 5)) +
  scale_y_log10()
ggsave("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section3_mutSig_dynamics_subtype/subtype_feature_association_analysis/subtype_snv_burden_boxplot.pdf")
```



### bxp2

<img src="..\..\images_for_markdown\Common_plots\boxplot\ordered_boxplot.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[mut_clonality_cli_association.R](G:\xiaolonggeのhome\已完成课题\克隆性分型课题\改稿\投Neuropathology and Applied Neurobiology\负责问题\code\submit\mut_clonality_cli_association.R)

```R
# clonality vs grade in IDHmut-codel

ggplot(filter(comb_plot_data, !is.na(neoplasm_histologic_grade), IDH.codel.subtype %in% "IDHmut-codel"), aes(neoplasm_histologic_grade, mut_num)) +
  geom_point(aes(col = CI95.timing), position = position_jitterdodge(jitter.width = 0.5, dodge.width = 1), shape = 16, alpha = 7/10, size = 4) + 
  geom_boxplot(fill = NA, position = position_dodge(width = 1), outlier.color = NA) +
  labs(x = NULL, y = "Number of mutations", col = NULL) +
  scale_color_manual(values = c("#D45658", "#3A84BB")) +
  scale_x_discrete(labels = c("Grade II", "Grade III")) +
  facet_grid(. ~ CI95.timing) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10, angle = 45, hjust = 0.5, vjust = 0.5), axis.ticks.x = element_blank(), 
        axis.title.y = element_text(size = 12), strip.background = element_blank(), strip.text = element_blank(), 
        panel.grid = element_blank(), legend.position = "top")
ggsave("/Users/jianlongliao/xiaolonggeのhome/已完成课题/克隆性分型课题/改稿/投Neuropathology and Applied Neurobiology/负责问题/section_2/grade_clonality_IDHmut-codel.pdf", width = 8, height = 8)
```



### bxp3

<img src="..\..\images_for_markdown\Common_plots\boxplot\ordered_boxplot_grid.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[mut_clonality_cli_association.R](G:\xiaolonggeのhome\已完成课题\克隆性分型课题\改稿\投Neuropathology and Applied Neurobiology\负责问题\code\submit\mut_clonality_cli_association.R)

```R
# clonality vs age in IDHmut-codel, IDHmut-non-codel and IDHwt gliomas

age_comb_plot_data <- comb_plot_data %>% 
  filter(!is.na(age_group)) %>%
  mutate(age_group = factor(age_group, levels = c("<=30", "(30,40]", "(40,50]", "(50,60]", "(60,70]", ">70")), 
         age_group = fct_recode(age_group, "30-40" = "(30,40]", "40-50" = "(40,50]", "50-60" = "(50,60]", "60-70" = "(60,70]"))

ggplot(age_comb_plot_data, aes(age_group, mut_num)) +
  geom_point(aes(col = age_group), position = position_jitterdodge(jitter.width = 2, dodge.width = 1), shape = 16, alpha = 5/10) + 
  geom_boxplot(fill = NA, position = position_dodge(width = 1), outlier.color = NA) +
  labs(x = "Age at diagnosis", y = "Number of mutations", fill = NULL) +
  scale_color_nejm() +
  coord_cartesian(ylim = c(0, 100)) +
  facet_grid(CI95.timing ~ IDH.codel.subtype, scales = "free_y") +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10, angle = 45, vjust = 0.5), axis.ticks.x = element_blank(), 
        axis.title.y = element_text(size = 12), strip.background = element_blank(), 
        panel.grid = element_blank(), legend.position = "none")
ggsave("/Users/jianlongliao/xiaolonggeのhome/已完成课题/克隆性分型课题/改稿/投Neuropathology and Applied Neurobiology/负责问题/section_2/age_clonality_limited_.pdf")
```



### bxp4

<img src="..\..\images_for_markdown\Common_plots\boxplot\ordered_boxplot_style2.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[mut_clonality_cli_association.R](G:\xiaolonggeのhome\已完成课题\克隆性分型课题\改稿\投Neuropathology and Applied Neurobiology\负责问题\code\submit\mut_clonality_cli_association.R)

```R
# clonality vs age in grade IV of IDHwt gliomas

ggplot(filter(age_comb_plot_data, !is.na(age_group), WHO_subtype %in% "IDHwt_grade_IV"), aes(age_group, mut_num)) +
  geom_point(aes(col = age_group), position = position_jitterdodge(jitter.width = 2, dodge.width = 1), shape = 16, alpha = 5/10) + 
  geom_boxplot(fill = NA, position = position_dodge(width = 1), outlier.color = NA) +
  labs(x = NULL, y = "Number of mutations", fill = NULL) +
  scale_color_nejm() +
  facet_grid(. ~ CI95.timing) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10, angle = 45, vjust = 0.5), axis.ticks.x = element_blank(), 
        axis.title.y = element_text(size = 12), strip.background = element_blank(),  
        panel.grid = element_blank(), legend.position = "none")
ggsave("/Users/jianlongliao/xiaolonggeのhome/已完成课题/克隆性分型课题/改稿/投Neuropathology and Applied Neurobiology/负责问题/section_2/age_clonality_IDHwt_gradeIV_.pdf")
```



## HeatMap

### hm1

<img src="..\..\images_for_markdown\Common_plots\heatmap\heatmap_style1.png"/>

#### 语言

R

#### 绘图模块或包

ComplexHeatmap, circlize

#### 代码

1. [mutSig_subtype_determination.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section3_mutSig_dynamics_subtype\code\mutSig_subtype_determination.R)

```R
# OncoSign cluster热图 ------------------------------------------------------

load("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Data/Processed_Data/mutation_data/SKCM_mut_data.Rdata")
skcm_sam_snv <- mut_data %>% 
  filter(Variant_Type %in% "SNP") %>% 
  count(bcr_patient_barcode) %>% 
  mutate(log_n = log10(n))

cluster_anno_data <- skcm_osc_cluster %>% 
  mutate(bcr_patient_barcode = str_sub(Sample, 1, 12)) %>%
  left_join(skcm_sam_snv, by = "bcr_patient_barcode") %>%
  left_join(skcm_comb_cli, by = "bcr_patient_barcode") %>%
  select(Sample:log_n, gender, age = age_at_initial_pathologic_diagnosis) %>%
  group_by(Osc_cluster) %>%
  arrange(n, .by_group = TRUE)

subtype_anno_top <- HeatmapAnnotation(`Total SNVs` = anno_barplot(cluster_anno_data$n, gp = gpar(fill = "#8491B4", col = "#8491B4")),
                                      Gender = cluster_anno_data$gender, 
                                      `Age at diagnosis` = cluster_anno_data$age, 
                                      Subtype = cluster_anno_data$Osc_cluster, 
                                      annotation_height = unit(c(2.5, 0.5, 0.5, 0.5), 'cm'), 
                                      col = list(Gender = c("FEMALE" = "#BB0021", "MALE" = "#80796B"), 
                                                 `Age at diagnosis` = colorRamp2(c(0, 100), c("white", "#B24745")), 
                                                 Subtype = c("ESC.1" = "#E64B35", "ESC.2" = "#4DBBD5", "ESC.3" = "#00A087", "ESC.4" = "#3C5488", "ESC.5" = "#F39B7F")))

sigs_label_heatmap_mat <- sigs_label_data_prep %>%
  semi_join(sigs_label_data_count, by = "Signature") %>%
  pivot_wider(names_from = Sample, values_from = label) %>%
  select(-Signature) %>% 
  as.matrix() %>% 
  set_rownames(sigs_label_data_prep_mat$Signature)
sigs_label_heatmap_mat <- sigs_label_heatmap_mat[, cluster_anno_data$Sample]

pdf("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section3_mutSig_dynamics_subtype/run_oncosign/skcm_sigs_oncosign_heatmap.pdf", width = 10, height = 5)
Heatmap(sigs_label_heatmap_mat, show_heatmap_legend = FALSE, 
        show_column_names = FALSE, 
        cluster_rows = FALSE, 
        cluster_columns = FALSE, 
        col = colorRamp2(c(-1, 0, 1), c("#0072B5", "gray96", "#BC3C29")), 
        top_annotation = subtype_anno_top)
dev.off()
```



### hm2

<img src="..\..\images_for_markdown\Common_plots\heatmap\heatmap_style2.png"/>

#### 语言

R

#### 绘图模块或包

ComplexHeatmap, circlize

#### 代码

1. PC端，[early_late_mutSig_cluster.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section2_mutSig_dynamics\code\early_late_mutSig_cluster.R)

```R
# signature贡献度变化聚类热图 ------------------------------------------------------

el_vary_con_sam_label <- tibble(Sample = colnames(el_vary_con)) %>% left_join(cluster_label_tib, by = "Sample")
col_anno <- HeatmapAnnotation(cluster = el_vary_con_sam_label$Cluster)

pdf("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section2_mutSig_dynamics/mutSig_dynamics/early_late_vary_heatmap.pdf", width = 10, height = 5)
Heatmap(el_vary_con, name = " ", 
        show_column_names = FALSE, 
        cluster_rows = FALSE, 
        cluster_columns = early_late_vary_consensus_cluster_result[[11]][["consensusTree"]], 
        col = colorRamp2(c(-0.4, 0, 0.4), c("#0072B5", "white", "#BC3C29")), 
        column_dend_height = unit(3, "cm"), 
        top_annotation = col_anno)
dev.off()
```



### hm3

<img src="..\..\images_for_markdown\Common_plots\heatmap\heatmap_style3.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[mutSig_subtype_therapy_analysis.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section3_mutSig_dynamics_subtype\code\mutSig_subtype_therapy_analysis.R)

```R
# 亚型间可靶向基因亚克隆比例热图 ---------------------------------------------------------

gene_sub_cluster_count_label <- target_gene_comb %>%
  count(Osc_cluster, Hugo_Symbol, ccf.ci95.clonality) %>%
  pivot_wider(names_from = ccf.ci95.clonality, values_from = n) %>%
  mutate(Clonal = ifelse(is.na(Clonal), 0, Clonal), Subclonal = ifelse(is.na(Subclonal), 0, Subclonal), 
         Sub_frac = Subclonal/(Clonal + Subclonal), Sub_sum_label = str_c(Subclonal, Clonal + Subclonal, sep = "/"))

sub_frac_diff_gene <- gene_sub_cluster_count_label %>%
  mutate(Hugo_Symbol = fct_inorder(Hugo_Symbol)) %>% # 防止split改变因子顺序从而与后续的unique不一致
  split(.$Hugo_Symbol) %>%
  map(~tibble(count = nrow(.), sub_frac_diff = psych::describe(.$Sub_frac)$range)) %>%
  bind_rows() %>%
  mutate(Hugo_Symbol = unique(gene_sub_cluster_count_label$Hugo_Symbol)) %>%
  filter(count > 1, sub_frac_diff > 0.4)

diff_gene_sub_frac_data <- gene_sub_cluster_count_label %>% semi_join(sub_frac_diff_gene, by = "Hugo_Symbol")
write_tsv(diff_gene_sub_frac_data, "/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section3_mutSig_dynamics_subtype/subtype_therapy_analysis/diff_gene_sub_frac_data.tsv")

ggplot(diff_gene_sub_frac_data, aes(Osc_cluster, Hugo_Symbol)) +
  geom_tile(aes(fill = Sub_frac)) +
  geom_text(aes(label = Sub_sum_label), size = 3) +
  labs(x = NULL, y = NULL, fill = "subclonal fraction") +
  scale_x_discrete(expand = c(0, 0)) +
  scale_fill_gradient(low = "white", high = "#006CB1", limits = c(0, 0.5), na.value = "#006CB1", breaks = c(0, 0.25, 0.5), labels = c("0.00", "0.25", ">0.50")) +
  ggthemes::theme_few() +
  theme(axis.ticks = element_blank(), axis.text.x = element_text(face = "bold", size = 10), axis.text.y = element_text(face = "italic", size = 10), 
        panel.background = element_rect(fill = "#C7C4C1"), panel.border = element_rect(colour = "black", size = 1))
ggsave("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section3_mutSig_dynamics_subtype/subtype_therapy_analysis/diff_gene_sub_frac_heatmap.pdf", width = 7, height = 10)
```



### hm4

<img src="..\..\images_for_markdown\Common_plots\heatmap\heatmap_style4.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[mutSig_vary_trajectory.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section2_mutSig_dynamics\code\mutSig_vary_trajectory.R)

```R
# 整体时间点signature贡献比例热图 ----------------------------------------------------

ggplot(skcm_mutSig_trajectory_comb, aes(Time_point, Sample)) +
  geom_tile(aes(fill = Weights)) +
  labs(y = NULL) +
  theme(axis.ticks.y = element_blank(), axis.text.y = element_blank()) +
  facet_grid(. ~ Signature)
ggsave("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section2_mutSig_dynamics/mutSig_dynamics/mutSig_trajectory/mutSig_trajectory_heatmap.pdf", width = 8, height = 10)
```



## LinePlot

### lp1

<img src="..\..\images_for_markdown\Common_plots\linePlot\dot_line_plot.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[mutSig_vary_trajectory.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section2_mutSig_dynamics\code\mutSig_vary_trajectory.R)

```R
# 代表性样本贡献度随时间变化点线图 --------------------------------------------------------

skcm_mutSig_trajectory_comb %>% 
  filter(Sample %in% repre_sample) %>%
  mutate(Signature = factor(Signature, levels = str_c("Signature.", c(1, 11, 17, 5, 7)))) %>%
  ggplot(aes(Time_point, Weights, col = Signature)) +
  geom_point() +
  geom_line(aes(group = Signature), size = 1.5) +
  labs(col = NULL) +
  facet_grid(Sample ~ .) +
  scale_color_nejm() +
  theme(legend.position = "top")
ggsave("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section2_mutSig_dynamics/mutSig_dynamics/mutSig_trajectory/represent_sample_mutSig_trac_line.pdf", width = 8, height = 10)
```



### lp2

<img src="..\..\images_for_markdown\Common_plots\linePlot\forest_line_plot.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[mutSig_vary_trajectory.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section2_mutSig_dynamics\code\mutSig_vary_trajectory.R)

```R
# 绘制平均signature贡献度随时间变化点线图 ------------------------------------------------

sig_time_point_stat %>% ungroup() %>%
  mutate(Signature = factor(Signature, levels = str_c("Signature.", c(1, 11, 17, 5, 7))), weight_min = mean_weight - sd_weight, weight_max = mean_weight + sd_weight) %>%
  mutate(weight_min = ifelse(weight_min >= 0, weight_min, 0)) %>%
  ggplot(aes(Time_point, mean_weight, ymin = weight_min, ymax = weight_max)) +
  # geom_errorbar(aes(col = Signature), size = 1.5) +
  geom_pointrange(aes(col = Signature), size = 1) +
  geom_line(aes(group = Signature, col = Signature), size = 1) +
  labs(y = "weight", col = NULL) +
  theme(legend.position = "top") +
  scale_color_nejm()
ggsave("/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section2_mutSig_dynamics/mutSig_dynamics/mutSig_trajectory/mean_sample_mutSig_trac_line.pdf")
```



## ForestPlot

### fp1

<img src="..\..\images_for_markdown\Common_plots\forestPlot\forestplot_baseR.png"/>

#### 语言

R

#### 绘图模块或包

base

#### 代码

1. PC端，[ccf_earlyLate_stat_plot.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section2_mutSig_dynamics\code\ccf_earlyLate_stat_plot.R)

```R
# ccf_forest_plot ---------------------------------------------------------
# 绘制突变基因的ccf森林图

# 癌基因克隆性统计

driver_clonality_stat <- skcm_ns_dri_ccf_data %>%
  count(Hugo_Symbol, ccf.ci95.clonality) %>%
  group_by(Hugo_Symbol) %>%
  mutate(fraction = n/sum(n))
write_tsv(driver_clonality_stat, "/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section2_mutSig_dynamics/ccf_infer_result/driver_clonality_stat.tsv")

ccf_forest_plot <- function(ccf_data, 
                            gene_col = "Hugo_Symbol", 
                            ccf_col = "absolute.ccf", 
                            ccf_upper_col = "absolute.ccf.0.95", 
                            ccf_lower_col = "absolute.ccf.0.05", 
                            clonality_col = "ccf.ci95.clonality", 
                            col = c("#BC3C29", "#0072B5"), 
                            plot_path){
  # ccf_data：突变ccf数据
  # gene_col：基因列名
  # ccf_col：ccf值列名
  # ccf_upper_col：散点图中值位点颜色
  # ccf_lower_col：是否对突变数目进行log10转换
  # clonality_col：
  # col：突变颜色
  # plot_path：绘图存放路径
  
  # ccf_data <- skcm_ns_dri_ccf_data
  # gene_col <- "Hugo_Symbol"
  # ccf_col <- "absolute.ccf"
  # ccf_upper_col <- "absolute.ccf.0.95"
  # ccf_lower_col <- "absolute.ccf.0.05"
  # clonality_col <- "ccf.ci95.clonality"
  # col <- c("#BC3C29", "#0072B5")
  
  ccf_data <- ccf_data %>% select(Gene = gene_col, ccf = ccf_col, ccf_up = ccf_upper_col, ccf_low = ccf_lower_col, clonality = clonality_col)
  
  ccf_data_ord1 <- ccf_data %>% filter(ccf < 1) %>% arrange(ccf)
  ccf_data_ord2 <- ccf_data %>% filter(ccf >= 1) %>% arrange(ccf_low)
  ccf_data_ord <- bind_rows(ccf_data_ord1, ccf_data_ord2)
  
  ccf_data_lt <- split(ccf_data_ord, ccf_data_ord$Gene)
  
  # list中存放每个signature的排序好的突变ccf数据，设定递增的x轴坐标(0-1之间，1-2之间...)
  
  plot_dat <- lapply(1:length(ccf_data_lt), function(i) {
    x <- ccf_data_lt[[i]] %>%
      mutate(scale_x_value = seq(i - 1, i, length.out = n()))
  })
  plot_dat_comb <- bind_rows(plot_dat)
  
  pdf(str_c(plot_path, "/ccf_forest_plot.pdf"))
  par(mar = c(5, 4, 0.5, 3))
  
  # 散点图的框架 ------------------------------------------------------------------
  
  plot(NA, NA, xlim = c(0, length(plot_dat)), ylim = c(0, 1), xlab = NA, ylab = "Cancer cell fraction", bty = "n", xaxt = "n", las = 2)

  # 散点和线条 ----------------------------------------------------------------------
  
  plot_dat_comb_split <- split(plot_dat_comb, plot_dat_comb$clonality)
  
  points(plot_dat_comb_split[[1]]$scale_x_value, plot_dat_comb_split[[1]]$ccf, pch = 16, cex = 0.4, col = col[1])
  points(plot_dat_comb_split[[2]]$scale_x_value, plot_dat_comb_split[[2]]$ccf, pch = 16, cex = 0.4, col = col[2])
  
  segments(x0 = plot_dat_comb_split[[1]]$scale_x_value, x1 = plot_dat_comb_split[[1]]$scale_x_value, y0 = plot_dat_comb_split[[1]]$ccf_low, y1 = plot_dat_comb_split[[1]]$ccf_up, col = adjustcolor(col = col[1], alpha.f = 0.4))
  segments(x0 = plot_dat_comb_split[[2]]$scale_x_value, x1 = plot_dat_comb_split[[2]]$scale_x_value, y0 = plot_dat_comb_split[[2]]$ccf_low, y1 = plot_dat_comb_split[[2]]$ccf_up, col = adjustcolor(col = col[2], alpha.f = 0.4))
  
  # x轴和y轴标签 -----------------------------------------------------------------
  
  axis(side = 1, at = seq(0.5, length(plot_dat) - 0.5, 1), labels = names(ccf_data_lt), las = 2, tick = TRUE)
  
 # 图例 ----------------------------------------------------------------------

  legend("right", c("Clonal", "Subclonal"), pch = 16, cex = 0.5, lty = 1, lwd=2, col = col, bty = "n", xpd = TRUE, text.width = 0.1, x.intersp = 0.6)
  
  dev.off()
}

skcm_dri_ccf_forest_plot <- ccf_forest_plot(skcm_ns_dri_ccf_data, plot_path = "/Users/jianlongliao/百度云同步盘/科研/graduation_project_master_2020/Section2_mutSig_dynamics/ccf_infer_result")
```



### fp2

<img src="..\..\images_for_markdown\Common_plots\forestPlot\forestplot_ggplot2.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[clonality_method_determination.R](G:\xiaolonggeのhome\已完成课题\克隆性分型课题\改稿\投Neuropathology and Applied Neurobiology\负责问题\code\submit\clonality_method_determination.R)

```R
# ccf confidence interval forest plot

glioma_dri_clonality_data %>%
  mutate(CI95_ccf0.8_compare = ifelse(CI95.timing == ccf.0.8, CI95.timing, "inconsistent")) %>%
  ggplot(aes(fct_reorder(mutation_id, absolute.ccf.0.95), ymin = absolute.ccf.0.05, ymax = absolute.ccf.0.95)) +
  geom_linerange(size = 0.05, aes(col = CI95.timing)) + 
  geom_hline(aes(yintercept = 0.8), colour="#BB0000", linetype="dashed") +
  geom_point(aes(y = absolute.ccf, col = CI95_ccf0.8_compare)) +
  annotate("text", x = 20, y = 0.85, hjust = 0, vjust = 1, label = "CCF = 0.8") +
  scale_color_manual(values = c("#D45658", "#C77CFF", "#3A84BB")) +
  labs(x = NULL, y = "Cancer cell fraction", col = NULL) +
  theme_bw() +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), panel.grid = element_blank(), legend.position = c(1, 0), legend.justification = c(1, 0), legend.background = element_rect(fill = NA), legend.key = element_rect(fill = NA))
ggsave("/Users/jianlongliao/xiaolonggeのhome/已完成课题/克隆性分型课题/改稿/投Neuropathology and Applied Neurobiology/负责问题/section_1/clonality_method_forest_plot.pdf", width = 6, height = 5)
```



## PiePlot

### pp1

<img src="..\..\images_for_markdown\Common_plots\PiePlot\scatterPie_tree_plot.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2, scatterpie, ggrepel

#### 代码

1. PC端，[mut_clonality_stat_analysis.R](G:\xiaolonggeのhome\已完成课题\克隆性分型课题\改稿\投Neuropathology and Applied Neurobiology\负责问题\code\submit\mut_clonality_stat_analysis.R)

```R
# cancer_gene_sub_frac_scatterpie

scatterpei_plot <- function(scatterpie_data, y_axis_data){
  
  plot <- ggplot(scatterpie_data) +
    scatterpie::geom_scatterpie(aes(pie_locus, mean_ccf, r = sqrt(sqrt(sqrt(mut_sam_num))) * 0.015), data = scatterpie_data, cols = c("Clonal", "Subclonal"), alpha = 0.9, color = NA) +
    ggrepel::geom_text_repel(aes(text_locus, mean_ccf, label = Hugo_Symbol), size = 1.2) +
    geom_vline(xintercept = 0) +
    geom_errorbar(aes(x = 0, y = mean_ccf, ymin = mean_ccf, ymax = mean_ccf), width = 0.05) +
    geom_errorbar(aes(x = 0, y = y_scale_value, ymin = y_scale_value, ymax = y_scale_value), width = 0.02, data = y_axis_data) +
    geom_text(aes(x = -0.025, y = y_scale_value, label = y_scale_value), data = y_axis_data, size = 1) +
    labs(x = "Mean of CCF", y = NULL, fill = NULL) +
    scale_fill_nejm() +
    scale_color_nejm() +
    coord_fixed() +
    theme_classic() +
    theme(legend.position = "none", axis.ticks = element_blank(), axis.text = element_blank(), axis.line.y = element_blank(), axis.title.x = element_text(size = 6), strip.background = element_blank(), strip.text = element_text(size = 6)) +
    facet_grid(. ~ IDH.codel.subtype)
  return(plot)
}

cancer_gene_scatterpie_data <- glioma_mut_clonality_data %>%
  filter(!Variant_Classification %in% c("3'Flank", "5'Flank", "3'UTR", "5'UTR", "Intron", "RNA", "Silent"), driver_mut %in% "Yes") %>%
  group_by(IDH.codel.subtype, Hugo_Symbol) %>%
  summarise(mut_sam_num = n_distinct(patient), mean_ccf = mean(absolute.ccf)) %>%
  arrange(desc(mean_ccf), .by_group = TRUE) %>%
  left_join(sam_stat, by = "IDH.codel.subtype") %>%
  mutate(mut_sam_frac = mut_sam_num / clonality_sample_num, x_locus = 1, pie_locus = x_locus * c(0.1, -0.1), text_locus = x_locus * c(0.25, -0.25)) %>%
  left_join(cancer_gene_sub_stat, by = c("IDH.codel.subtype", "Hugo_Symbol"))

y_axis_data <- tibble(y_scale_value = seq(0.3, 1, by = 0.1))

cancer_gene_scatterpie_selected_data <- cancer_gene_scatterpie_data %>%
  filter(mut_sam_frac > 0.02, mean_ccf > 0.6)

y_axis_selected_data <- tibble(y_scale_value = seq(0.6, 1, by = 0.1))

cancer_gene_scatterpie <- scatterpei_plot(cancer_gene_scatterpie_data, y_axis_data)
cancer_gene_scatterpie_selected <- scatterpei_plot(cancer_gene_scatterpie_selected_data, y_axis_selected_data)

ggsave("/Users/jianlongliao/xiaolonggeのhome/已完成课题/克隆性分型课题/改稿/投Neuropathology and Applied Neurobiology/负责问题/section_1/cancer_gene_sub_frac_scatterpie.pdf", cancer_gene_scatterpie)
ggsave("/Users/jianlongliao/xiaolonggeのhome/已完成课题/克隆性分型课题/改稿/投Neuropathology and Applied Neurobiology/负责问题/section_1/cancer_gene_sub_frac_scatterpie_selected.pdf", cancer_gene_scatterpie_selected)
```



## SurvivalPlot

### sup1

<img src="..\..\images_for_markdown\Common_plots\survivalPlot\survival_curve.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2

#### 代码

1. PC端，[mutSig_subtype_survival_analysis.R](G:\xiaolonggeのhome\毕业前的洗礼\2020硕士毕业\graduation_project_master_2020\Section3_mutSig_dynamics_subtype\code\mutSig_subtype_survival_analysis.R)

```R
# OncoSign cluster生存曲线差异 ----------------------------------------------------

refine_survival_plot <- function(surv_data, sur_type_col, sur_time_col, sam_group_col, y_lab_name){
  
  # surv_data <- skcm_osc_cluster_sur
  # sur_type_col <- "OS"
  # sur_time_col <- "OS.time"
  # sam_group_col <- "Osc_cluster"
  # y_lab_name <- "Overall survival (OS) probability"
  
  fit_result <- surv_fit(Surv(surv_data[[sur_time_col]], surv_data[[sur_type_col]]) ~ surv_data[[sam_group_col]], data = surv_data)
  
  ori_os_plot <- ggsurvplot(fit_result, pval = TRUE, ggtheme = theme_bw())
  refine_os_plot <- ori_os_plot$plot + 
    labs(y = y_lab_name, col = NULL) + 
    scale_color_nejm(label = str_remove(names(fit_result$strata), "surv_data\\[\\[sam_group_col\\]\\]="))  + 
    theme(legend.position = c(1, 1), legend.justification = c(1, 1), legend.background = element_rect(fill = NA), legend.key = element_rect(fill = NA))
}

dynamic_label_os_plot <- refine_survival_plot(skcm_osc_cluster_sur, "OS", "OS.time", "Dynamic_label", "Overall survival (OS) probability")
```



## CombPlot

### cp1

<img src="..\..\images_for_markdown\Common_plots\combPlot\ordered_barplot_comb.png"/>

#### 语言

R

#### 绘图模块或包

ggplot2, ggsci

#### 代码

1. 阿里云盘，[mutsig_fit.R](https://www.aliyundrive.com/drive/folder/5fb620ec327d6290953342bbb1cf37392d746f2f)
2. PC端，[mutsig_fit.R](F:\common_R_functions\mutsig_fit.R)

```R
sam_we_plot <- function(mutSig_we_tib, rm_sample = NULL, plot_path){
  # mutSig_we_tib：signature在样本中的weight及exposure数据
  # rm_sample：为便于图形展示而删除的样本
  # plot_path：绘图存放路径
  
  # mutSig_we_tib <- skcm_mutSig_mer_list$mutSig_we_pass
  # rm_sample <- "TCGA-FW-A3R5-06"
  
  if(!is.null(rm_sample)){
    mutSig_we_tib <- filter(mutSig_we_tib, !Sample %in% rm_sample)
  }
  
  mutSig_we_tib <- filter(mutSig_we_tib, Weights > 0)
  sam_total_ep <- mutSig_we_tib %>%
    count(Sample, wt = Exposure, name = "Mut_count") %>%
    arrange(Mut_count)
  
  mutSig_we_tib$Sample <- factor(mutSig_we_tib$Sample, levels = sam_total_ep$Sample)
  
  mutSig_ep_plot <- ggplot(mutSig_we_tib, aes(x = Sample, y = Exposure, fill = Signature)) +
    geom_bar(stat = "identity") +
    labs(x = "", y = "Number of mutations", fill = "") + 
    guides(fill = guide_legend(nrow = 1)) + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), legend.position = 'top') + 
    scale_fill_nejm()
  
  mutSig_wt_plot <- ggplot(mutSig_we_tib, aes(x = Sample, y = Weights, fill = Signature)) +
    geom_bar(stat = "identity") +
    labs(x = "", y = "Fraction of signatures") + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), legend.position = "none") + 
    scale_fill_nejm()
  
  pdf(str_c(plot_path, '/mutSig_we_barplot.pdf'), width = 10, height = 6)
  grid.arrange(mutSig_ep_plot, mutSig_wt_plot, ncol =1, nrow = 2, heights = c(2, 3))
  dev.off()
  
}
```



### cp2

<img src="..\..\images_for_markdown\Common_plots\combPlot\scatterPlot_barplot_comb.png"/>

#### 语言

R

#### 绘图模块或包

base

#### 代码

1. 阿里云盘，[mutsig_fit.R](https://www.aliyundrive.com/drive/folder/5fb620ec327d6290953342bbb1cf37392d746f2f)
2. PC端，[mutsig_fit.R](F:\common_R_functions\mutsig_fit.R)

```R
mutSig_ep_plot <- function(mutSig_tib, col = NULL, bg_col = c("#EDF8B1", "#2C7FB8"), median_col = "red", logscale = TRUE, plot_path){
  # mutSig_tib：signature在样本中的exposure数据
  # col：为每个signature指定颜色，若NULL则为灰色
  # bg_col：散点图的背景颜色
  # median_col：散点图中值位点颜色
  # logscale：是否对突变数目进行log10转换
  # plot_path：绘图存放路径
  
  # mutSig_tib <- skcm_mutSig_mer_list$mutSig_we_pass
  # col <- mutSig_col
  # bg_col <- c("#EDF8B1", "#2C7FB8")
  # median_col <- "red"
  # logscale <- TRUE
  
  total_sam <- n_distinct(mutSig_tib$Sample)
  
  mutSig_ep <- mutSig_tib %>%
    filter(Signature != "Unknown", Exposure > 0) %>%
    select(-Weights)
  
  mutSig_ep_med <- mutSig_ep %>%
    group_by(Signature) %>%
    summarise(Case = n(), Case_frac = Case/total_sam, Ep_med = median(Exposure), Ep_med_log10 = log10(Ep_med)) %>%
    arrange(Ep_med)
  
  mutSig_ep$Signature <- factor(x = mutSig_ep$Signature, levels = mutSig_ep_med$Signature)  
  mutSig_ep_lt <- split(mutSig_ep, mutSig_ep$Signature)
  
  # list中存放每个signature的排序好的exposure数据，设定递增的x轴坐标(0-1之间，1-2之间...)
  
  plot_dat <- lapply(1:length(mutSig_ep_lt), function(i) {
    x <- mutSig_ep_lt[[i]] %>%
      arrange(Exposure) %>%
      mutate(scale_x_value = seq(i - 1, i, length.out = n()))
  })
  names(plot_dat) = names(mutSig_ep_lt)
  
  # 限制y轴长度 ------------------------------------------------------------------
  
  if (logscale) {
    y_lims = range(log10(bind_rows(plot_dat)$Exposure))
  }else {
    y_lims = range(bind_rows(plot_dat)$Exposure)
  }
  y_max = ceiling(max(y_lims))
  y_min = floor(min(y_lims))
  y_lims = c(y_min, y_max)
  y_at = pretty(y_lims) # y轴虚线分割线
  
  pdf(str_c(plot_path, "/mutSig_ep_prev.pdf"))
  par(mar = c(0.5, 4, 0.5, 1), mfrow = c(2, 1))
  
  # 条形图 ---------------------------------------------------------------------
  
  barplot(mutSig_ep_med$Case_frac, col = col, border = NA, ylim = c(0, 1), las = 2)
  mtext(text = "Fraction of samples", side = 2, line = 2.4)
  
  # 散点图的框架 ------------------------------------------------------------------
  
  plot(NA, NA, xlim = c(0, length(plot_dat)), ylim = y_lims, xlab = NA, ylab = NA, bty = "n", xaxt = "n", las = 2)
  rect(xleft = seq(0, length(plot_dat) - 1, 1), ybottom = y_min, xright = seq(1, length(plot_dat), 1), 
       ytop = y_max, col = grDevices::adjustcolor(col = bg_col, alpha.f = 0.2), border = NA)
  abline(h = pretty(y_lims), lty = 2, col = "gray70")
  
  # 散点 ----------------------------------------------------------------------
  
  if (is.null(col)) {
    col <- rep("gray70", length(plot_dat))
    names(col) <- names(plot_dat)
  }
  
  lapply(1:length(plot_dat), function(i) {
    x = plot_dat[[i]]
    if (logscale) {
      points(x$scale_x_value, log10(x$Exposure), pch = 16, cex = 0.4, col = col[names(plot_dat)[i]])
    }else {
      points(x$scale_x_value, x$Exposure, pch = 16, cex = 0.4, col = col[names(plot_dat)[i]])
    }
  })
  
  # x轴和y轴标签 -----------------------------------------------------------------
  
  axis(side = 1, at = seq(0.5, length(plot_dat) - 0.5, 1), labels = str_sub(names(plot_dat), start = 11), las = 1, tick = FALSE, line = -1.5)
  
  if (logscale) {
    mtext(text = "Number of mutations (log10)", side = 2, line = 2)
  }else {
    mtext(text = "Number of mutations", side = 2, line = 2)
  }
  
  # 中值线 ---------------------------------------------------------------------
  
  if (logscale) {
    lapply(1:nrow(mutSig_ep_med), function(i) {
      x_seq <- quantile(c(i -1, i), seq(0.1, 1, 0.1))
      segments(x0 = x_seq["40%"], x1 = x_seq["60%"], y0 = mutSig_ep_med$Ep_med_log10[i], y1 = mutSig_ep_med$Ep_med_log10[i], col = median_col)
    })
  }else {
    lapply(1:nrow(mutSig_ep_med), function(i) {
      x_seq <- quantile(c(i -1, i), seq(0.1, 1, 0.1))
      segments(x0 = x_seq["40%"], x1 = x_seq["60%"], y0 = mutSig_ep_med$Ep_med[i], y1 = mutSig_ep_med$Ep_med[i], col = median_col)
    })
  }
  
  dev.off()
  
}
```



